### ISSUE TITLE
GU0035 should not warn here

### ISSUE THREAD

**Issue by JohanLarsson** [JohanLarsson/Gu.Analyzers#59]
```c#
using System;

public sealed class Foo
{
    private static readonly IDisposable Cached = new Disposable();
    private readonly IDisposable value;

    private Foo(IDisposable value)
    {
        this.value = value;
    }

    public static Foo Create() => new Foo(Cached);
}
```
